### ISSUE TITLE
Invalid front matter causes 500 server error

### ISSUE THREAD

**Issue by djuber** [forem/forem#16624]
**Describe the bug**

When I have the Basic Markdown editor (editor v1)
When creating a new post with this content  (rather than hash keys, a plain string, in the frontmatter)

```yml
---
Text in the frontmatter?
---

Not a good idea
``` 

When I preview the post, I get a 500 server error (and no indication why).

**To Reproduce**
visit /settings/customization and confirm Basic Markdown Editor Version is selected
click Create Post
Erase the frontmatter template (title/ cover image, tags) and replace with a simple string.
Click the preview tab.

**Expected behavior**

Either the frontmatter delimiter dashes `---` are treated as horizontal rules when no key-value pairs are provided, or an indication that the frontmatter is invalid is shown to the user.

**Screenshots**

![Preview Error](https://user-images.githubusercontent.com/1237369/154511845-e56ab34c-8631-4c4a-a386-9de304ff1bf5.png)

**Additional context**

https://app.honeybadger.io/projects/66984/faults/84085022 (no method `to_h` for String at `parsed.front_matter.to_h` in `ArticlesController#preview`)

Started after #16569 fixed another error showing on article preview

**Comment by citizen428** [comment]
This problem seems to (partially) originate in `Article#has_frontmatter?`:

```ruby
  def has_frontmatter?
    fixed_body_markdown = MarkdownProcessor::Fixer::FixAll.call(body_markdown)
    begin
      parsed = FrontMatterParser::Parser.new(:md).call(fixed_body_markdown)
      parsed.front_matter["title"].present?
    rescue Psych::SyntaxError, Psych::DisallowedClass
      # if frontmatter is invalid, still render editor with errors instead of 500ing
      true
    end
  end
```

BTW, when trying to post instead of previewing you'll see different errors:

![Screen Shot 2022-02-21 at 11 29 45](https://user-images.githubusercontent.com/47985/154889371-f550052e-fc95-4a1e-84ac-385bb102bb2d.png)

**Comment by mikeydorje** [comment]
Closing as the 500 error is now handled gracefully in the preview action. If you can still reproduce a server error, please open a new issue with current steps.
